### PR TITLE
Add async and sync support to metadata handler

### PR DESCRIPTION
**Pull request by everaldorodrigo**

### CODE REVIEW

File: biothings/web/handlers/query.py [hunk 1]
```diff
@@ -149,10 +150,14 @@ async def get(self):
...
-        meta = self.extras(meta)  # override here
+        if (iscoroutinefunction(self.extras)):
+            meta = await self.extras(meta)
+        else:
+            meta = self.extras(meta)
+
         self.finish(dict(sorted(meta.items())))
 
-    def extras(self, _meta):
+    async def extras(self, _meta):
```
Comment (newgene) [inline biothings/web/handlers/query.py:160]: @everaldorodrigo if `extra` can be sync or async, shall we need the original method unchanged as a sync method? Or is there any reason to make it an async method by default?

Comment (everaldorodrigo) [reply biothings/web/handlers/query.py:160]: > @everaldorodrigo if `extra` can be sync or async, shall we need the original method unchanged as a sync method? Or is there any reason to make it an async method by default?

@newgene, there's no strong reason in making it async by default. We could keep it sync.

Comment (newgene) [reply biothings/web/handlers/query.py:160]: 👍 let's keep it unchanged then, then the subclass can override it as async method if needed.

Comment (everaldorodrigo) [reply biothings/web/handlers/query.py]: :done: :bowtie:

**Pull request merged**